lua/luasnip/extras/postfix.lua

```diff
@@ -55,6 +55,19 @@ local function generate_opts(match_pattern, user_callback)
 	}
 end
 
+local function wrap_condition(user_condition, match_pattern)
+	return function(line_to_cursor, matched_trigger, captures)
+		return (
+			not user_condition
+			or user_condition(line_to_cursor, matched_trigger, captures)
+		)
```

Comment:
Yup :+1:
I think it would be better to do the check for `user_condition` once while constructing the function, and not every time it is called, wdyt?

Reply:
> Could you also work in an option to disable our condition? Should be fine to pass it via context, no need for yet another opts-table xD

I didn't want to add another option 😂

Can't we do one of the following?
1. If user provides `condition` just use that instead of the internal one
2. Let use return `nil` from the provided `condition` to fallback to the internal one

1 is very simple and clean. 2 gives the user to handle specialized cases and fallback to the internal logic when they want (by returning `nil` instead of `boolean`).

---

If you definitely want to add another option, just give me a name for it (naming is hard 😂) and I can add it in.

Reply:
> I think it would be better to do the check for user_condition once while constructing the function, and not every time it is called, wdyt?

If we check `user_condition` outside the actual callback, we either need to:
- assign a fallback function that always returns `true` to `user_condition`
- or return another callback with only `return line_to_cursor:sub(1, -1 - #matched_trigger):match(match_pattern) ~= nil`.

Which one do you prefer?

Reply:
> Which one do you prefer?

I guess the first one will look a bit better in the code.. There's a `yes`-function in util, `require("luasnip.util.util").yes`


> I didn't want to add another option :joy:

I do get that xD
But I think it's the best way to optionally turn off our condition I think

> Can't we do one of the following?
> 
> 1. If user provides `condition` just use that instead of the internal one
> 2. Let use return `nil` from the provided `condition` to fallback to the internal one
> 
> 1 is very simple and clean. 2 gives the user to handle specialized cases and fallback to the internal logic when they want (by returning `nil` instead of `boolean`).
> 
> If you definitely want to add another option, just give me a name for it (naming is hard joy) and I can add it in.

Ahh what I don't really like there is that "I want to expand postfix regardless of what is in front of it" is unrelated to wanting a `condition`, so it's a bit weird to require a condition to get that desired behaviour.

Reply:
> I want to expand postfix regardless of what is in front of it

That can be controlled by tweaking `match_pattern` right? Using `match_pattern = [[[%w%.%_%-%"%']*$]]` would enable user to expand postfix even if there is nothing in front of it.

Reply:
Ouuuuhh yeah that's cool :+1: